homework.md

```diff
@@ -44,70 +44,72 @@ Using CSS, how would I select every `<h1>` tag on the page?
 
 What is JavaScript?
 
-A programming language used on the internet
+JavaScript, often abbreviated as JS, is a programming language that conforms to the ECMAScript specification.A programming language used on the internet.
 
 ## Question 8
 
 What three problem does Git & Github solve?
 
-When people want to show off code to each other they can put it on Github
+1.Helpes us to share information.
+2.Enables people to make thier own changes to that information,and share it back.
+3.It allows  us to track how inforamtion has changed between each checkpoint and to go back to the older versions of the work if we to see what we tried before or to undo changes.
 
 ## Question 9
 
 What happens when you `fork` a repository?
 
-You delete it
+A fork is a copy of a repository. Forking a repository allows you to freely experiment with changes without affecting the original project.
 
 ## Question 10 
 
 What happens when you clone a repostory?
 
-It send it to a friend
+The "clone" command downloads an existing Git repository to your local computer.
 
 ## Question 11
 
 What is a Pull Request?
 
-When you send a file over the internet
+Pull requests let you tell others about changes you've pushed to a branch in a repository on GitHub.
 
 ## Question 12
 
 What is the Git History log?
 
-It lets you see into the future
+git log lists the commits made in that repository in reverse chronological order.
 
 ## Question 13
 
 What does it meant to "push" changes to Github?
 
-You send the changes of your local repository to Github
+When you made a change on GitHub, it not only saved the change and committed the file at the same time.
 
 ## Question 14
 
 What is Github?
 
-A place for us to store and share our code.
+GitHub, Inc. is a subsidiary of Microsoft which provides hosting for software development and version control using Git. It offers the distributed version control and source code management functionality of Git, plus its own features.
 
 ## Question 15
 
 What does it mean to "Commit your changes"?
 
-Send you code to Github
+If you have made changes to the same file in both your local copy(on your harddrive) of a Git repository and a remote repository, you'll encounter the “commit your changes or stash them before you can merge.
```

Comment:
A simpler understanding may be that it is how you save your work before "pushing" it - would you agree?

Reply:
Noted. I've simplified my answer.